tests/SecureFileExtensionTest.php

```diff
@@ -46,4 +46,17 @@ public function testDefaultPermissions() {
 		$this->assertTrue($subfolder->canView());
 	}
 
+	public function testRestrictedDefaultPermissions() {
+		$folder = $this->objFromFixture('Folder', 'default-root');
+		Session::set('loggedInAs', null);
+		$this->assertTrue($folder->canView());
+
+		Config::inst()->update('SecureAssets', 'Defaults', array('Permission' => 'LoggedInUsers'));
+		$this->assertFalse($folder->canView());
+
+		$subfolder = $this->objFromFixture('Folder', 'default-child');
+		$this->assertEquals('Inherit', $subfolder->CanViewType);
+		$this->assertFalse($subfolder->canView());
+	}
```

Comment:
Can you add a test for Default permissions OnlyTheseUsers, add a group, and test that for two users (one in and one out of the group), only the one in the right group can access root folder with Inherit? (Phew, what a mouthful).